search/src/RawImage.cpp

```diff
@@ -169,39 +143,43 @@ void RawImage::applyMask(int flags, const std::vector<int>& exceptions, const Ra
    (which is how the code is generally used. If you are only
    growing the mask by 1, the extra copy will be a little slower.
 */
-void RawImage::growMask(int steps) {
-    const int num_pixels = width * height;
-
-    // Set up the initial masked vector that stores the number of steps
-    // each pixel is from a masked pixel in the original image.
-    std::vector<int> masked(num_pixels, -1);
-    for (int i = 0; i < num_pixels; ++i) {
-        if (pixels[i] == NO_DATA) masked[i] = 0;
-    }
+void RawImage::growMask(int steps, bool on_gpu) {
+    if (on_gpu) {
+        deviceGrowMask(width, height, pixels.data(), pixels.data(), steps);
+    } else {
```

Comment:
To avoid adding another nesting layer, just break out early for gpu processing. i.e.

```
if(on_gpu){
  deviceGrowMask();
  return;
}
//keep everything else the same.
```

Reply:
Done.